pkg/cephmgr/osd/device.go

```diff
@@ -212,28 +213,53 @@ func partitionBluestoreMetadata(context *clusterd.Context, info *partition.Metad
 	return nil
 }
 
-// Partitions a device for use by a bluestore osd.
+// Partitions a device for use by a osd.
 // If there are any partitions or formatting already on the device, it will be wiped.
-func partitionBluestoreOSD(context *clusterd.Context, config *osdConfig) error {
-	blockDetails, err := getBlockPartitionDetails(config)
+func partitionOSD(context *clusterd.Context, config *osdConfig) error {
+	dataDetails, err := getDataPartitionDetails(config)
 	if err != nil {
 		return err
 	}
 
 	// zap/clear all existing partitions on the device
-	err = sys.RemovePartitions(blockDetails.Device, context.Executor)
+	err = sys.RemovePartitions(dataDetails.Device, context.Executor)
 	if err != nil {
-		return fmt.Errorf("failed to zap partitions on metadata device /dev/%s: %+v", blockDetails.Device, err)
+		return fmt.Errorf("failed to zap partitions on metadata device /dev/%s: %+v", dataDetails.Device, err)
 	}
 
 	// create the partitions on the device
-	err = sys.CreatePartitions(blockDetails.Device, config.partitionScheme.GetPartitionArgs(), context.Executor)
+	err = sys.CreatePartitions(dataDetails.Device, config.partitionScheme.GetPartitionArgs(), context.Executor)
 	if err != nil {
-		return fmt.Errorf("failed to partition /dev/%s. %+v", blockDetails.Device, err)
+		return fmt.Errorf("failed to partition /dev/%s. %+v", dataDetails.Device, err)
+	}
+
+	if config.partitionScheme.StoreType == Filestore {
+		// the OSD is using filestore, create a filesystem for the device and mount it under config root
+		dataPartDetails := config.partitionScheme.Partitions[FilestoreDataPartitionName]
+		dataPartPath := filepath.Join(diskByPartUUID, dataPartDetails.PartitionUUID)
+		logger.Infof("waiting for partition path %s", dataPartPath)
+		err = waitForPath(dataPartPath, context.Executor)
+		if err != nil {
+			return fmt.Errorf("failed waiting for %s: %+v", dataPartPath, err)
+		}
+		err = sys.FormatDevice(dataPartPath, context.Executor)
+		if err != nil {
+			logger.Warningf("first attempt to format partition %s on device %s failed.  Waiting 2 seconds then retrying: %+v",
```

Comment:
The retry is right below on line 250, that's where we call `sys.FormatDevice` a second time.